pkg/build/api/validation/validation.go

```diff
@@ -272,15 +259,6 @@ func validateBinarySource(source *buildapi.BinaryBuildSource) fielderrors.Valida
 	return allErrs
 }
 
-func validateRevision(revision *buildapi.SourceRevision) fielderrors.ValidationErrorList {
-	allErrs := fielderrors.ValidationErrorList{}
-	if len(revision.Type) == 0 {
-		allErrs = append(allErrs, fielderrors.NewFieldRequired("type"))
-	}
-	// TODO: validate other stuff
```

Comment:
there's really not much that can be validated on it, SourceRevision has one struct, GitSourceRevision, and that's basically all string fields with no particular requirements.  about the only thing that could be validated is that the email address looks like an email address, and even that isn't really a requirement (it won't break us if it's not one).